core/src/main/java/aima/core/search/basic/support/StraightLineDistanceHeuristic.java

```diff
@@ -0,0 +1,42 @@
+package aima.core.search.basic.support;
+
+import aima.core.environment.map2d.GoAction;
+import aima.core.environment.map2d.InState;
+import aima.core.environment.map2d.Map2D;
+import aima.core.search.api.Node;
+import aima.core.util.datastructure.Point2D;
+
+import java.util.Arrays;
+import java.util.function.ToDoubleFunction;
+
+/**
+ * @author wormi
+ */
```

Comment:
This is specific to 'GoAction' and 'InState' so does not belong within the basic.support package. To re-use I would recommend moving it from InformedSearchTest to ProblemFactory, and declaring it as a public static class (I will likely make this change).